### DISCUSSION TITLE
Inverted (mirror) text on TFT shield for Arduino ili9486 parallel

### DISCUSSION THREAD

**Discussion by visionSensor** [Bodmer/TFT_eSPI#3110] [New processor/display support or capability request]
Hello, I have an issue with a TFT shield for Arduino.
Screen controller: ili9486
Microcontroller: ESP32
IDE: Arduino

Library version TFT_espi: 2.5.43
Data bus: 8-bit parallel
Problem: I have a 3.5-inch TFT shield (480X320) that uses an 8-bit parallel bus, ili9486 driver. The TFT works well with the ESP32, with good examples and colors. However, the problem arises when I display text with a certain screen rotation (setRotation(0);). The text appears inverted or mirrored.
I'm not sure if it's visible, but inverted text is appearing at the top of the screen:

![rainbow](https://github.com/Bodmer/TFT_eSPI/assets/155329826/29cad2a6-9a4e-41f3-8f6b-68488fdcf9ab)

Here is the result of a sketch that I wrote, in which I think the problem is more noticeable:

![hello groop](https://github.com/Bodmer/TFT_eSPI/assets/155329826/26a51812-5fa6-4f54-ae5c-630fb92d96c4)

"Back of the TFT shield:

![back tft](https://github.com/Bodmer/TFT_eSPI/assets/155329826/5ab1afa1-fc48-4ac5-a48b-c0d8c2f0bec6)

Help me, please, Bodmer.

**Comment by Bodmer** [comment]
Provide a link to the display vendors page. The driver may not be an ILI9486.

Alternatively if may just be another configuration variant, in which case you may need to adjust the flip/mirror settings in the init code here:
https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_Drivers/ILI9486_Rotation.h

You may find that setRotation with values 4 to 7 corrects the problem.

**Comment by visionSensor** [comment]
Hello Bodmer, thank you for the suggestion to try with the rotation (setRotation()). I tested it with this code in the Arduino IDE:

**Comment by visionSensor** [comment]
#include <SPI.h>
#include <TFT_eSPI.h>
TFT_eSPI tft = TFT_eSPI();  
void setup() {
  // put your setup code here, to run once:
tft.init();
tft.fillScreen(TFT_BLACK);
tft.setRotation(0);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(1);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(2);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(3);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(4);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(5);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(6);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
tft.fillScreen(TFT_BLACK);
tft.setRotation(7);
tft.setCursor(100,100);
tft.setTextColor(TFT_WHITE);
tft.setTextSize(1);
tft.println("Hello World!");
delay(1000);
}

void loop() {
  // put your main code here, to run repeatedly:

}

**Comment by visionSensor** [comment]
The result, the text rotates, but it seems that only in the horizontal position, the text is displayed correctly.

![rotation](https://github.com/Bodmer/TFT_eSPI/assets/155329826/b819bd10-6289-4d89-8a01-d7ed1b77721f)

**Comment by Bodmer** [comment]
It seems you have a variant of the displays I have.

The TFT_MAD_MX/V/Y bits change the way the pixels are written to the display.  I have not seen correctly rendered characters written in reverse before so do check with the vendor that you have an ILI9486 display.

Unfortunately the only option is to try different settings in the rotation code.

**Reply by visionSensor** [reply]
Here is the link to the store where I bought the screen. Well, I actually purchased the screen at the physical store, but you get the idea:https://saisac.pe/producto/pantalla-3-5-para-arduino-uno-mega-de-480x320-copia/

**Reply by jasonc1025-333** [reply]
Thanks @Bodmer for an amazing TFT_eSPI library to utilize the numerous: ili9486 and ST7735's (all found on Amazon 😉) as inexpensive displays for Jr High Robotics.  Also, thanks @visionSensor for creating this helpful Post. 🙂
